packages/lodestar/src/sync/initial/initial.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * @module sync/initial
+ */
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+import {IBeaconChain} from "../../chain";
+import {ReputationStore} from "../IReputation";
+import {IReqResp} from "../../network";
+import {ILogger} from "../../logger";
+import {ISyncOptions} from "../options";
+import {IInitialSyncModules, InitialSyncEventEmitter} from "./interface";
+import {EventEmitter} from "events";
+import {getTargetEpoch, isValidHeaderChain} from "../utils/sync";
+import {BeaconState, Checkpoint} from "@chainsafe/eth2.0-types";
+import {computeStartSlotOfEpoch} from "../../chain/stateTransition/util";
+import {getBlockRange} from "../utils/blocks";
+
+export class InitialSync
+  extends (EventEmitter as { new(): InitialSyncEventEmitter })
+  implements InitialSyncEventEmitter {
+
+  private config: IBeaconConfig;
+  private opts: ISyncOptions;
+  private chain: IBeaconChain;
+  private reps: ReputationStore;
+  private rpc: IReqResp;
+  private logger: ILogger;
+  private peers: PeerInfo[];
+
+  public constructor(opts: ISyncOptions, {config, chain, network, reps, logger, peers}: IInitialSyncModules) {
+    super();
+    this.config = config;
+    this.chain = chain;
+    this.peers = peers;
+    this.reps = reps;
+    this.opts = opts;
+    this.rpc = network.reqResp;
+    this.logger = logger;
+  }
+
+  public async start(): Promise<void> {
+    this.logger.info("initial sync start");
+    if(this.peers.length === 0) {
+      this.logger.error("No peers. Exiting initial sync");
+      return;
+    }
+    if(!this.chain.isInitialized()) {
+      this.logger.warn("Chain not initialized.");
+      this.emit("sync:completed", null);
+      return;
+    }
+
+    const chainCheckPoint = this.chain.latestState.currentJustifiedCheckpoint;
+    //listen on finalization events
+    this.chain.on("processedEpoch", this.sync);
+    //start syncing from current chain checkpoint
+    await this.sync(chainCheckPoint);
+  }
+
+  public async stop(): Promise<void> {
+    this.logger.info("initial sync stop");
+    this.chain.removeListener("processedEpoch", this.sync);
+  }
+
+  private sync = async (chainCheckPoint: Checkpoint) => {
```

Comment:
sync = async 😂 Nothing wrong here, just think its funny haha

Reply:
can we call this fastSync, or somehow differentiate it from a naive sync?
I think we probably will have support for several initial syncing strategies (eg: fast sync, naive sync, light sync, etc)